### DISCUSSION TITLE
Report of resource consumption

### DISCUSSION THREAD

**Discussion by MFJI** [Xilinx/finn#830] [Q&A] [answered]
Hello all,

I found that the resource consumption from the FINN report is more than the resources that the board I set has. Does anyone know what is the reason for this? Thanks.

**Comment by MFJI** [comment]
I am using the board alveo U200, and I found that there is no board file of the alveo U200 in Vivado 2022.1. Is this the thing that causes the problem? Is the report in step "step_generate_estimate_reports" serious enough?

And also, is it possible for me to decide the type of RAM to use? For example, I would like to use BRAMs instead of URAMs?

**Comment by auphelia** [comment, marked as answer]
Hi @MFJI ,

The resources that FINN reports can be on three different levels:
- FINN resource estimations: Calculates resource estimations based on [FINN-R: An End-to-End Deep-Learning Framework for Fast Exploration of Quantized Neural Networks](https://arxiv.org/abs/1809.04570)
- HLS resources: Resource estimations from the Vitis HLS synthesis
- Post synthesis resources: Resources reported by Vivado synthesis

I assume you have used the resource estimation builder step, which generates resource estimations before any synthesis was run. As the name says, these are "estimations" and might over-estimate your actual resource usage.

You can specify which memory to use by specifying the [ram_style](https://github.com/Xilinx/finn/blob/main/src/finn/custom_op/fpgadataflow/matrixvectoractivation.py#L102) node attribute. It can be specified using the `folding_config.json`. You can find an example here: https://github.com/Xilinx/finn/blob/main/src/finn/qnn-data/build_dataflow/folding_config.json

The options are:
- `auto` -> let Vivado/Vitis decide
- `block` -> BRAM
- `distributed` -> LUTRAM
- `ultra` -> URAM
